jscomp/ml/parmatch.ml

```diff
@@ -1106,8 +1106,8 @@ let build_other ext env : Typedtree.pattern = match env with
       0L Int64.succ p env
 | ({pat_desc=(Tpat_constant (Const_bigint _))} as p,_) :: _ ->
     build_other_constant
-      (function Tpat_constant(Const_bigint i) -> String.length i | _ -> assert false)
-      (function i -> Tpat_constant(Const_bigint (String.make i '*')))
+      (function Tpat_constant(Const_bigint (sign, i)) -> String.length (Bigint_utils.to_string sign i) | _ -> assert false)
+      (function i -> Tpat_constant(Const_bigint (true, (String.make i '*'))))
```

Comment:
not sure what `(String.make i '*')` is doing here?
I would have guessed `string_of_int i` instead.
Not sure how to trigger this code path, perhaps with some example?

Reply:
Not sure. I guess this function is used in pattern matching for gadt partially. But logically I think we can change `(String.make i '*')` to `string_of_int i` here for less space complexity.